extensions/windows-event-log/wel/MetadataWalker.cpp

```diff
@@ -90,10 +86,11 @@ bool MetadataWalker::for_each(pugi::xml_node &node) {
     if (it != formatFlagMap.end()) {
       std::function<std::string(const std::string &)> updateFunc = [&](const std::string &input) -> std::string {
         if (resolve_) {
-          auto resolved = windows_event_log_metadata_.getEventData(it->second);
-          if (!resolved.empty()) {
-            return resolved;
+          const auto resolved = windows_event_log_metadata_.getEventData(it->second);
+          if (resolved && !resolved->empty()) {
+            return *resolved;
           }
+          // TODO(szaszm): add error logging
```

Comment:
One behavior change in `getEventData`: allocation failure now returns an `error_code`, instead of an empty string. Since both empty strings and errors are ignored here, there is no change in behavior. Added a comment to do proper error handling later, but I didn't wanna do it in the context of this change. (`MetadataWalker` has no logger.)